### DISCUSSION TITLE
让xmake可以使用add_links 给msvc link .dll.a 文件

### DISCUSSION THREAD

**Discussion by cherichy** [xmake-io/xmake#2938] [Ideas]
### Xmake 版本

2.7.2

### 操作系统版本和架构

Windows 11

### 描述问题

对于有些工程（如julia），需要使用msvc 链接mingw编译的.dll.a 导入库。
直接在VS里面是可以在附加导入库填入导入库全称的，如libjulia.dll.a
在使用xmake的时候，由于识别到用的msvc，会自动在add_links 给的名字后面加 .lib
所以会报错 error: LINK : fatal error LNK1181: 无法打开输入文件“libjulia.dll.a.lib”
 

### 期待的结果

让add_links可以使用文件的全称，如在windows下可以使用 xxx.lib libxxx.dll.a

**Comment by waruqi** [comment]
add_ldflags("libjulia.dll.a", {force = true})

**Reply by cherichy** [reply]
谢谢！
